controller/node_controller.go

```diff
@@ -357,27 +357,45 @@ func (nc *NodeController) syncDiskStatus(node *longhorn.Node) error {
 		}
 		// get disk available size
 		diskInfo, err := nc.getDiskInfoHandler(disk.Path)
+		readyCondition := types.GetDiskConditionFromStatus(diskStatus, types.DiskConditionTypeReady)
 		// if the file system has changed
 		if err != nil {
-			logrus.Errorf("Get disk information on node %v error: %v", node.Name, err)
+			if readyCondition.Status != types.ConditionStatusFalse {
+				readyCondition.LastTransitionTime = util.Now()
+			}
+			readyCondition.Status = types.ConditionStatusFalse
+			readyCondition.Reason = types.DiskConditionReasonDiskIsNotExist
+			readyCondition.Message = fmt.Sprintf("Get disk information on node %v error: %v", node.Name, err)
 			// disable invalid disk
 			updateDisk.AllowScheduling = false
 			updateDisk.StorageMaximum = 0
 			updateDisk.StorageReserved = 0
 			diskStatus.StorageAvailable = 0
 		} else if diskInfo == nil || diskInfo.Fsid != diskID {
-			logrus.Errorf("disk %v on node %v has changed file system", disk.Path, node.Name)
+			if readyCondition.Status != types.ConditionStatusFalse {
+				readyCondition.LastTransitionTime = util.Now()
+			}
+			readyCondition.Status = types.ConditionStatusFalse
+			readyCondition.Reason = types.DiskConditionReasonDiskHasGone
+			readyCondition.Message = fmt.Sprintf("disk %v on node %v has changed file system", disk.Path, node.Name)
 			// disable invalid disk
 			updateDisk.AllowScheduling = false
 			updateDisk.StorageMaximum = 0
```

Comment:
Realized StorageMaximum should be in disk.Status instead of disk.Spec. It's a status we detected.

Filed https://github.com/rancher/longhorn/issues/212 . Can be addressed in a separate PR.